src/test_fixture/app.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+    error::Error,
+    ff::Serializable,
+    helpers::{
+        query::{QueryConfig, QueryInput},
+        ByteArrStream,
+    },
+    secret_sharing::IntoShares,
+    test_fixture::network::InMemoryNetwork,
+    AppSetup, HelperApp,
+};
+
+use crate::test_fixture::network::InMemoryTransport;
+use futures_util::future::try_join_all;
+use generic_array::GenericArray;
+use typenum::Unsigned;
+
+pub trait IntoBuf {
+    fn into_buf(self) -> Vec<u8>;
+}
+
+impl<I, S> IntoBuf for I
+where
+    I: IntoIterator<Item = S>,
+    S: Serializable,
+{
+    fn into_buf(self) -> Vec<u8> {
+        let this = self.into_iter();
+        let (lb, ub) = this.size_hint();
+        let cnt = ub.unwrap_or(lb);
```

Comment:
This seems a bit sketchy to me.  If your upper bound is `None` or short of the true size of the input, you will allocate  something that is too small and then overrun the buffer (and panic).  I think that we want an additional constraint: `I::IntoIter: ExactSizeIterator`, so that you can just do:

```suggestion
        let cnt = self.len();
```